src/controllers/sensors.controller.ts

```diff
@@ -0,0 +1,108 @@
+import { UUIDV4 } from 'sequelize';
+import { sensorsDB } from '../models/db.index';
+import { Sensor } from '../models/sensors.model';
+import { RequestHandler } from 'express';
+
+export const getSensors: RequestHandler = async (req, res) => {
+    try {
+        const sensors = await sensorsDB.findAll().then((sensors) => sensors.map((sensors) => {
+            const jsonSensors:Sensor = sensors.toJSON();
+            return jsonSensors;
+        })) as Sensor[];
+        res.status(200).json({ sensors });
+    } catch (e) {
+        res.status(500).json({ message: e });
+    }
+};
+
+export const getSensor: RequestHandler = async (req, res) => {
+    try {
+        const id = req.params.id;
+        if (!id) {
+            res.status(400).json({ message: 'Missing required fields' });
+            return;
+        }
+        const sensor = await sensorsDB.findOne({ where: { id: id } }).then((sensor) => {
+            const jsonSensor:Sensor|undefined = sensor?.toJSON();
+            return jsonSensor;
+        });
+        if (!sensor){
+            res.status(400).json({ message: 'Sensor not found' });
+            return;
+        }
+        res.status(200).json({ sensor });
+    } catch (e) {
+        res.status(500).json({ message: e });
+    }
+};
+
+interface CreateSensorBody {
+    sensor:Partial<Sensor>;
```

Comment:
We are leaving this partial, but which fields are allowed to be omitted? Maybe we type this more explicitly, as well as check in the route if req fields are missing and reply with an error if so.

Reply:
also do we want to check for auth? verifying a usertype is what they said earlier?